File: src/optical_element/MATLAB/g_Furion_ellipsoid_Mirror.m
classdef  g_Furion_ellipsoid_Mirror < g_Furion_cylinder_ellipse_Mirror%追迹phase转换成角度
  properties 
   
  end
methods
function obj = g_Furion_ellipsoid_Mirror( beam_in,ds,di,chi,theta,surface,r1,r2,grating)   
    obj=obj@g_Furion_cylinder_ellipse_Mirror( beam_in,ds,di,chi,theta,surface,r1,r2,grating) ;%输入参数给父类   
    
end
end

methods
    function set_center(obj, beam_in,ds,di,chi,theta,surface,r1,r2,grating)
        obj.center =g_ellipsoid( beam_in,ds,di,chi,theta,surface,r1,r2,grating);      
    end
end
end

File: src/optical_element/g_Furion_ellipsoid_Mirror.cpp
#include "g_Furion_ellipsoid_Mirror.h"
#include "g_ellipsoid.h"

using namespace Furion_NS;

G_Furion_ellipsoid_Mirror::G_Furion_ellipsoid_Mirror(G_Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r1, double r2, Grating* grating)
    : G_Furion_Cylinder_Ellipse_Mirror(beam_in, ds, di, chi, theta, surface, r1, r2, grating)
{
    cout << "G_Furion_ellipsoid_Mirror 初始化" << endl;
    //test();
}

G_Furion_ellipsoid_Mirror::~G_Furion_ellipsoid_Mirror()
{
    //delete center;
    //cout << "G_Furion_ellipsoid_Mirror 析构" << endl;
}

void G_Furion_ellipsoid_Mirror::set_center(G_Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r1, double r2, Grating* grating)
{
    center = new G_Ellipsoid(beam_in, ds, di, chi, theta, surface, r1, r2, grating);
    cout << "G_Furion_ellipsoid_Mirror的set_center" << endl;
}